test_crates/enum_discriminants/src/lib.rs

```diff
@@ -0,0 +1,5 @@
+#[repr(C)]
+pub enum A {
+    One = 1,
+    Two = 1 + 1,
+}
```

Comment:
Consider adding some more test cases for discriminants that are defined implicitly.

For example:
```rust
pub enum Example {
    Zero,
    Hundred = 100,
    HundredAndOne,
}
```
has three defined discriminants, 0, 100, and 101, per the reference: https://doc.rust-lang.org/reference/items/enumerations.html#implicit-discriminants

Similarly, we want to check that implicit discriminants on tuple and struct enums work as well, as the reference allows those too if the enum has an explicit primitive repr.

Reply:
Can do this for explicit enumerations. Explain more about implicit enumerations in a comment below.

Reply:
Assuming you agree with my suggestion about changing the definition away from `"The discriminant, if explicitly specified."` to one where the edge exists if the discriminant is well-defined _via any mechanism_, it would be useful to have a test that we compute the implicit discriminants correctly in all cases.

For example:
```
pub enum Example {
    Zero,
    One,
    Two,
}
```
has well-defined discriminants: 0, 1, 2. They don't have to be specified explicitly here.